plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -415,6 +409,24 @@ def finalize(self):
             self.parent.repomd_file_context.finalize()
 
 
+class SignRepoMetadataStep(platform_steps.PluginStep):
+    def __init__(self, working_dir, step=constants.PUBLISH_SIGN_REPO_METADATA_STEP):
+        """
+        Initialize and set the ID of the step
+        """
+        super(SignRepoMetadataStep, self).__init__(step)
+        self.description = _("Signing repo metadata")
+        self.repomd = os.path.join(working_dir, REPO_DATA_DIR_NAME, REPOMD_FILE_NAME)
+
+    def finalize(self):
+        sign_options = configuration.get_gpg_sign_options(
+            self.get_repo(), self.get_config()
+        )
+        assert sign_options
+        signer = util.Signer(options=sign_options)
+        signer.sign(self.repomd)
```

Comment:
I had an impression that semantically in `finalize`, we usually perform some cleanup or closing actions and such. Here we are signing, basically performing the main action this step is for, so my natural approach would be to put it into `process_main`. Is it because this code was performed in the finalizing stage originally, you decided to keep it here?

Reply:
I was actually unsure about that because `process_main` seems to sometimes be used on a per-item basis.  I can rename it to `process_main` if that is the appropriate solution.

Reply:
I guess it's safer to leave it in `finalize`, since it was there before, and actually it would be more correct because finalization of other steps will be run by that time. If we put it in the process_main it might not happen in the correct order.
Anyway, you convinced me :) Let's keep it as you did it.